weblate/api/tests.py

```diff
@@ -1235,6 +1237,54 @@ def test_patch(self):
         )
         self.assertEqual(response.data["slug"], "new-slug")
 
+    def test_create_component_docfile(self):
+        with open(TEST_DOC, "rb") as handle:
+            response = self.do_request(
+                "api:project-components",
+                self.project_kwargs,
+                method="post",
+                code=201,
+                superuser=True,
+                request={
+                    "docfile": handle,
+                    "name": "Local project",
+                    "slug": "local-project",
+                    "repo": "local:",
+                    "vcs": "local",
+                    "filemask": "*.strings",
+                    "template": "en.strings",
+                    "file_format": "strings-utf8",
+                    "push": "https://username:password@github.com/example/push.git",
+                    "new_lang": "none",
+                },
+            )
+        self.assertEqual(response.data["repo"], "local:")
+        self.assertEqual(Component.objects.count(), 2)
+
+    def test_create_component_zipfile(self):
+        with open(TEST_ZIP, "rb") as handle:
+            response = self.do_request(
+                "api:project-components",
+                self.project_kwargs,
+                method="post",
+                code=201,
+                superuser=True,
+                request={
+                    "zipfile": handle,
+                    "name": "Local project",
+                    "slug": "local-project",
+                    "repo": "local:",
+                    "vcs": "local",
+                    "filemask": "*.strings",
+                    "template": "en.strings",
+                    "file_format": "strings-utf8",
```

Comment:
If this passes it shows lack of validation. The zip file doesn't contain these files.

The proper configuration should be:

```suggestion
                    "filemask": "*.po",
                    "new_base": "project.pot",
                    "file_format": "po",
```

But please keep this broken test as well and make sure the validation fails in this case - the configured files are not present in the zip file and the component clean method should fail.